tests/Unit/Domain/Test_Domain.cpp

```diff
@@ -0,0 +1,96 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <catch.hpp>
+#include <memory>
+
+#include "Domain/CoordinateMaps/AffineMap.hpp"
+#include "Domain/Domain.hpp"
+#include "Utilities/Gsl.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+namespace {
+template <size_t Dim>
+void test_domain_construction(
+    const Domain<Dim, Frame::Grid>& domain,
+    const std::vector<std::unordered_map<Direction<Dim>, BlockNeighbor<Dim>>>&
+        expected_block_neighbors,
+    const std::vector<std::unordered_set<Direction<Dim>>>&
+        expected_external_boundaries) {
+  const auto& blocks = domain.blocks();
+  for (size_t i = 0; i < blocks.size(); ++i) {
+    const auto& block = blocks[i];
+    CHECK(block.id() == i);
+    CHECK(block.neighbors() == expected_block_neighbors[i]);
+    CHECK(block.external_boundaries() == expected_external_boundaries[i]);
+  }
+}
+
+void test_1d_domains() {
+  {
+    // Test two block domain
+    auto two_block_affine_map = []() {
```

Comment:
Why is this a lambda?

Reply:
Yes, why not just pass the `std::make_unique` calls to the initializer_list constructor of vector?